File: src/main/java/architecture/goldenboughs_lib/client/model/item/GuiGeoItemModel.java
package architecture.goldenboughs_lib.client.model.item;

import architecture.goldenboughs_lib.client.model.GeoModelExpand;
import architecture.goldenboughs_lib.core.GoldenBoughsLib;
import net.minecraft.resources.ResourceLocation;
import software.bernie.geckolib.animatable.GeoAnimatable;

public class GuiGeoItemModel<T extends GeoAnimatable> extends GeoItemModel<T> {
	private static final ResourceLocation MODELLED_PATH = GeoModelExpand.model(GoldenBoughsLib.modRl("item/gui_item_model")).withPrefix(MODEL_SUFFIX);
	private final ResourceLocation resourceLocation;

	public GuiGeoItemModel(ResourceLocation path) {
		super(path);
		this.resourceLocation = path.withPrefix("textures/item/").withSuffix(".png");
	}

	public GuiGeoItemModel(ResourceLocation modelPath, ResourceLocation textureName, ResourceLocation animationsName) {
		super(modelPath, textureName, animationsName);
		this.resourceLocation = textureName.withPrefix("textures/item/").withSuffix(".png");
	}

	@Override
	public ResourceLocation getModelResource(T animatable) {
		return MODELLED_PATH;
	}

	@Override
	public ResourceLocation getTextureResource(T animatable) {
		return resourceLocation;
	}
}

File: src/main/kotlin/architecture/goldenboughs_lib/client/model/item/GuiGeoItemModel.kt
package architecture.goldenboughs_lib.client.model.item

import architecture.goldenboughs_lib.core.Lib
import net.minecraft.resources.ResourceLocation
import software.bernie.geckolib.animatable.GeoAnimatable

class GuiGeoItemModel<T : GeoAnimatable> : GeoItemModel<T> {
	private val texturesRl: ResourceLocation

	constructor(path: ResourceLocation) : super(path) {
		this.texturesRl = path.withPrefix("textures/item/").withSuffix(".png")
	}

	constructor(modelPath: ResourceLocation, textureName: ResourceLocation, animationsName: ResourceLocation) : super(
		modelPath,
		textureName,
		animationsName
	) {
		this.texturesRl = textureName.withPrefix("textures/item/").withSuffix(".png")
	}

	override fun getModelResource(animatable: T?): ResourceLocation {
		return MODELLED_PATH
	}

	override fun getTextureResource(animatable: T?): ResourceLocation {
		return texturesRl
	}

	companion object {
		private val MODELLED_PATH: ResourceLocation = Lib.modRl("geo/item/gui_item_model.geo.json")
	}
}
